app/src/main/kotlin/com/pitchedapps/frost/injectors/JsInjector.kt

```diff
@@ -106,4 +110,38 @@ fun FrostWebViewClient.jsInject(vararg injectors: InjectorContract) = web.jsInje
 class JsInjector(val function: String) : InjectorContract {
     override fun inject(webView: WebView) =
         webView.evaluateJavascript(function, null)
+}
+
+/**
+ * Helper object to obfuscate window tags for JS injection.
+ */
+private object TagObfuscator {
+
+    fun obfuscateTag(tag: String) : String {
+        val rnd = Random(tag.hashCode() + salt)
+        val obfuscated = StringBuilder()
+            .append(prefix)
+            .append(randomChars(rnd, tag.length))
+        L._d { "TagObfuscator: Obfuscating tag '$tag' to '$obfuscated'" }
+        //if (BuildConfig.DEBUG) {
+        //    return "_frost_${tag.toLowerCase(Locale.CANADA)}"
+        //} else
+        return obfuscated.toString()
+    }
+
+    private val salt by lazy { System.currentTimeMillis() }
+
+    private val prefix by lazy {
+        val rnd = Random(System.currentTimeMillis())
+        val length = rnd.nextInt(10, 16)
+        randomChars(rnd, length)
+    }
+
+    private fun randomChars(random: Random, count: Int) : String {
```

Comment:
I'm fine with the approach, but I don't like generating keys one at a time.
Perhaps you can use [big integer](https://stackoverflow.com/questions/2290057/how-to-generate-a-random-biginteger-value-in-java/2702630) and output a base64 output, or at the very least take in the existing stringbuilder and append to it rather than make a new string builder